server/src/main/java/org/opensearch/search/pipeline/PipelinedRequest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+
+/**
+ * Groups a search pipeline based on a request and the request after being transformed by the pipeline.
+ */
+public class PipelinedRequest {
+    private final Pipeline pipeline;
+    private final SearchRequest transformedRequest;
+
+    PipelinedRequest(Pipeline pipeline, SearchRequest transformedRequest) {
+        this.pipeline = pipeline;
+        this.transformedRequest = transformedRequest;
+    }
+
+    public SearchResponse transformResponse(SearchResponse response) {
+
+        // TODO: Should we transform responses with the original request or the transformed request? Or both?
```

Comment:
Is there a plan here? If we can't resolve this now, can we link to an issue?

Reply:
I'm going to make the decision that response processors will get the transformed request.

In future we may also decide to pass the original request, but then we can just add a default method to the interface that ignores it so we don't break existing response processors.

Reply:
Can we remove this TODO then?

Reply:
Nevermind, I wasn't looking at the latest revision